cylc/flow/config.py

```diff
@@ -573,6 +573,10 @@ def __init__(
 
         self.mem_log("config.py: end init config")
 
+        self.namespaces: Dict[str, List[str]] = {}
+        for name in self.taskdefs:
+            self.namespaces[name] = self.taskdefs[name].namespace_hierarchy
+
```

Comment:
Don't we already have this info in the config somewhere?

Reply:
Yes, got it.